### PR TITLE
Allow setting default location for weather widget

### PR DESCRIPTION
**Pull request by Mazhoon**
In cases where all of the users are from the same geographical area, it would be extremely useful to be able to set a default location for the dashboard weather widget. With these changes, this option becomes available, as default values can be set with occ config:app:set weather_status [lat,lon,altitude,address] --value 'value'

These changes also implement the path 2 described in feature request https://github.com/nextcloud/server/issues/27908 denied in 2021

Signed-off-by: Tuomas Nurmi <tuomas.nurmi@opinsys.fi>

### CODE REVIEW

File: apps/weather_status/lib/Service/WeatherStatusService.php [hunk 1]
```diff
@@ -330,9 +330,9 @@ private function searchForAddress(string $address): array {
 	 * @return array which contains coordinates, formatted address and current weather status mode
 	 */
 	public function getLocation(): array {
-		$lat = $this->config->getUserValue($this->userId, Application::APP_ID, 'lat', '');
-		$lon = $this->config->getUserValue($this->userId, Application::APP_ID, 'lon', '');
-		$address = $this->config->getUserValue($this->userId, Application::APP_ID, 'address', '');
+        $lat = $this->config->getUserValue($this->userId, Application::APP_ID, 'lat', $this->config->getAppValue(Application::APP_ID, 'lat', ''));
+        $lon = $this->config->getUserValue($this->userId, Application::APP_ID, 'lon', $this->config->getAppValue(Application::APP_ID, 'lat', ''));
```
Comment (come-nc) [inline apps/weather_status/lib/Service/WeatherStatusService.php:334]:
```suggestion
        $lon = $this->config->getUserValue($this->userId, Application::APP_ID, 'lon', $this->config->getAppValue(Application::APP_ID, 'lon', ''));
```

**Review by come-nc — COMMENTED** [review]
No strong opinion about this

**Comment by PVince81** [comment]
@jancborchardt thoughts on this ? note: this requires occ commands to set the values, there's no UI

**Comment by jancborchardt** [comment]
@skjnldsv it seems like a useful feature – is it very difficult to get merged, or was it closed for a specific reason? cc @sorbaugh as it was before your time.

**Comment by skjnldsv** [comment]
> @jancborchardt thoughts on this ? note: this requires occ commands to set the values, there's no UI

@jancborchardt Can you reply to this? 

Regarding the closure, this doesn't mean this is a bad idea, we're just cleaing old stale PR.
(though my comment came late because yesterday was public holiday, sorry for the confusion @Mazhoon :sun_with_face:)
That doesn't mean we do not want that feature, but it seems incomplete for now.

**Comment by skjnldsv** [comment]
If we want to stay code-only, (there isn't really an admin settings UI location), we could have this as a config option like the default timezone :thinking: 
https://github.com/nextcloud/server/blob/2ea6713504aa49d1a8cb9a7efaad8706484986c1/config/config.sample.php#L269

**Pull request closed without merging**